pytorch_lightning/utilities/cli.py

```diff
@@ -76,6 +83,64 @@ def add_lightning_class_args(
             instantiate=not issubclass(lightning_class, Trainer),
         )
 
+    def add_optimizer_args(
+        self,
+        optimizer_class: Union[Type[Optimizer], Tuple[Type[Optimizer], ...]],
+        nested_key: str = 'optimizer',
+        link_to: str = 'AUTOMATIC',
+    ) -> None:
+        """
+        Adds arguments from an Optimizer class to a nested key of the parser
+
+        Args:
+            optimizer_class: Any subclass of torch.optim.Optimizer.
+            nested_key: Name of the nested namespace to store arguments.
+            link_to: Dot notation of a parser key to set arguments or AUTOMATIC.
+        """
+        if isinstance(optimizer_class, tuple):
+            assert all(issubclass(o, Optimizer) for o in optimizer_class)
+        else:
+            assert issubclass(optimizer_class, Optimizer)
+        kwargs = {
+            'instantiate': False,
+            'fail_untyped': False,
+            'skip': {'params'},
+        }
+        if isinstance(optimizer_class, tuple):
+            self.add_subclass_arguments(optimizer_class, nested_key, required=True, **kwargs)
+        else:
+            self.add_class_arguments(optimizer_class, nested_key, **kwargs)
+        self.optimizers_and_lr_schedulers[nested_key] = (optimizer_class, link_to)
+
+    def add_lr_scheduler_args(
```

Comment:
How can the user configure dictionary for the scheduler as follow:

def configure_optimizers(self):
return optimizer, {"scheduler": scheduler, "internal": "step", "monitor": "val_loss", ...}

Reply:
Would be something like

```python
class MyModel(LightningModule):
    def __init__(self, optimizer_init: dict, lr_scheduler_init: dict):
        super().__init__()
        self.optimizer_init = optimizer_init
        self.lr_scheduler_init = lr_scheduler_init

    def configure_optimizers(self):
        optimizer = instantiate_class(self.parameters(), self.optimizer_init)
        scheduler = instantiate_class(optimizer, self.lr_scheduler_init)
        return optimizer, {"scheduler": scheduler, "internal": "step", "monitor": "val_loss", ...}

class MyLightningCLI(LightningCLI):
    def add_arguments_to_parser(self, parser):
        parser.add_optimizer_args(..., link_to='model.optimizer_init')
        parser.add_lr_scheduler_args(..., link_to='model.lr_scheduler_init')
```